cmd/trust-manager/app/app.go

```diff
@@ -103,18 +102,10 @@ func NewCommand() *cobra.Command {
 							},
 						},
 						&corev1.ConfigMap{}: {
-							// Only cache full ConfigMaps in the "watched" namespace.
-							Namespaces: map[string]cache.Config{
-								opts.Bundle.Namespace: {},
-							},
+							// Watch full config maps across all namespaces.
+							// TODO: create a seperate cache for targets and sources and only
+							// cache full config maps for the sources.
 						},
-						// Only cache partial metadata for targets.
-						&metav1.PartialObjectMetadata{
-							TypeMeta: metav1.TypeMeta{
-								APIVersion: "v1",
-								Kind:       "ConfigMap",
-							},
-						}: {},
```

Comment:
Does it work to set the `Namespaces` field on the cache options here? And instead remove the typed cache above? Wouldn't that fix the TODO, or am I not understanding how this works?

Reply:
We want to fully cache the configmaps in the "trusted namespace", but also want to cache the metadata of all configmaps across all namespaces.
It does not seem like that is achievable using the current API.

Reply:
Setting the Namespaces field will break the all-namespace metadata caching.

Reply:
Thanks for reminding me that the goal here is to manage configmaps throughout the cluster. 😊